python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -771,6 +770,19 @@ def my_dbt_op(dbt: DbtCliResource):
             manifest, dagster_dbt_translator = get_manifest_and_translator_from_dbt_assets(
                 [assets_def]
             )
+
+            dbt_resource_props_by_test_name = get_dbt_resource_props_by_test_name(manifest)
+            has_asset_check = any(
+                dbt_resource_props.get("meta", {}).get("dagster", {}).get("asset_check", False)
+                for dbt_resource_props in dbt_resource_props_by_test_name.values()
+            )
+
+            # When dbt is enabled with asset checks, we turn off any indirection with dbt selection.
+            # This way, the Dagster context completely determines what is executed in a dbt
+            # invocation with a subsetted selection.
+            if version.parse(dbt_version) >= version.parse("1.5.0") and has_asset_check:
```

Comment:
Sadly, in dbt 1.4, an empty `DBT_INDIRECT_SELECTION` is not supported: https://docs.getdbt.com/reference/node-selection/test-selection-examples#indirect-selection.